contracts/ligo/src/swaps/fa2_whitelisted_swap.mligo

```diff
@@ -0,0 +1,63 @@
+#import "../../fa2/fa2_interface.mligo" "FA2"
+#include "../../fa2_modules/admin/non_pausable_simple_admin.mligo"
+#include "fa2_swap.mligo"
+
+(* ======== Swaps whitelist component ======== *)
+
+(* ==== Types ==== *)
+
+type whitelist = (address, unit) big_map
+
+let addr_list_to_big_map(new_allowed: address list) : whitelist =
+  List.fold
+    (fun (m, a : (address, unit) big_map * address) -> Big_map.add a () m)
+    new_allowed
+    (Big_map.empty : (address, unit) big_map)
+
+let swap_check_whitelist(whitelist, swap_param : whitelist * swap_entrypoints) : unit =
+  match swap_param with
+    Start swap_offer ->
+      [@inline]
+      let check_asset(on_err : string)(assets : fa2_assets) : unit =
+        match Big_map.find_opt assets.fa2_address whitelist with
+        | Some u -> ()
+        | None -> failwith on_err
+        in
+      let u = List.map (check_asset "SWAP_OFFERED_FA2_NOT_WHITELISTED")
+                        swap_offer.assets_offered in
+      let u = List.map (check_asset "SWAP_REQUESTED_FA2_NOT_WHITELISTED")
+                        swap_offer.assets_requested in
+      ()
+  | Accept x -> unit
```

Comment:
A bit odd that in some places you use `unit` and in others `()`, I'd settle on the former and use that all the time.

Reply:
Right, agree here.